examples/model_selection/plot_grid_search_digits.py

```diff
@@ -13,56 +13,177 @@
+    precision_threshold = 0.98
+
+    cv_results_ = pd.DataFrame(cv_results)
+    print(f"\nModels with a precision higher than {precision_threshold}:")
+    print_dataframe(cv_results_)
+
+    # Filter-out all results below the threshold
+    high_precision_cv_results = cv_results_[
+        cv_results_["mean_test_precision"] > precision_threshold
+    ]
+
+    high_precision_cv_results = high_precision_cv_results[
+        [
+            "mean_score_time",
+            "mean_test_recall",
+            "std_test_recall",
+            "mean_test_precision",
+            "std_test_precision",
+            "rank_test_recall",
+            "rank_test_precision",
+            "params",
+        ]
+    ]
+
+    # Select the most performant models in terms of recall
+    # (within 1 sigma from the best)
+    best_recall_std = high_precision_cv_results["mean_test_recall"].std()
+    best_recall = high_precision_cv_results["mean_test_recall"].max()
+    best_recall_threshold = best_recall - best_recall_std
+
+    high_recall_cv_results = high_precision_cv_results[
+        high_precision_cv_results["mean_test_recall"] > best_recall_threshold
+    ]
+    print(
+        "Out of the previously selected high precision models, we find the highest\n"
+        "recall model and also keep other models without one standard deviation of\n"
+        "this model: "
+    )
+    print_dataframe(high_recall_cv_results)
+
+    # From the best candidates, select the fastest model to predict
+    fastest_top_recall_high_precision_index = high_recall_cv_results[
+        "mean_score_time"
+    ].idxmin()
+
+    print(
+        "\nThe selected final model is the fastest to predict out of the previously\n"
+        "selected subset of best models based on precision and recall.\n"
+        "Its scoring time is:\n\n"
+        f"{high_recall_cv_results.loc[fastest_top_recall_high_precision_index]}"
+    )
+
+    return fastest_top_recall_high_precision_index
+
+
+# %%
+# Once we defined our strategy to select the best model, we can create the
+# grid-search instance. Subsequently, we can check the best parameters found.
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+
+clf = GridSearchCV(SVC(), tuned_parameters, scoring=scores, refit=refit_strategy)
+
+# %%
+# Tuning hyper-parameters
+# -----------------------
+
+clf.fit(X_train, y_train)
+print(f"\nThe best set of parameters found are:\n{clf.best_params_}")
+
+# %%
+# Finally, we evaluate the fine-tuned model on the left-out evaluation set.
+from sklearn.metrics import classification_report
+
+y_pred = clf.predict(X_test)
+print(
+    "\nOur selected model will have the following performance on the "
+    f"evaluation set:\n\n {classification_report(y_test, y_pred)}"
+)
+
+# %%
+# .. note::
+#    The problem is too easy: the hyperparameter plateau is too flat and the
+#    output model is the same for precision and recall with ties in quality.
```

Comment:
SVM-based models perform well for this problem. I think using a simpler algorithm (e.g. `LogisticRegression`) might avoid models having nearly identical performances.